src/main/java/jep/Jep.java

```diff
@@ -497,103 +353,30 @@ public Object getValue(String str) throws JepException {
     private native Object getValue(long tstate, String str, Class<?> clazz)
             throws JepException;
 
-    /**
-     * @deprecated use Python 3 bytes object instead and
-     *             {@link #getValue(String,Class)} with byte[].class
-     *
-     *             Retrieves a Python string object as a Java byte[].
-     * 
-     * @param str
-     *            the name of the Python variable to get from the
-     *            sub-interpreter's global scope
-     * @return an <code>Object</code> array
-     * @throws JepException
-     *             if an error occurs
-     */
-    @Deprecated
-    public byte[] getValue_bytearray(String str) throws JepException {
-        isValidThread();
-
-        return getValue_bytearray(this.tstate, str);
-    }
-
-    private native byte[] getValue_bytearray(long tstate, String str)
-            throws JepException;
-
-    private native long createModule(long tstate, String name)
-            throws JepException;
-
     // -------------------------------------------------- set things
 
-    /**
-     * Sets the default classloader.
-     * 
-     * @deprecated This may be removed in a future version of Jep. Jep does not
-     *             fully support changing the ClassLoader after construction.
-     * 
-     * @param cl
-     *            a <code>ClassLoader</code> value
-     */
-    @Deprecated
-    public void setClassLoader(ClassLoader cl) {
-        if (cl != null && cl != this.classLoader) {
-            this.classLoader = cl;
-            // call native set
-            setClassLoader(this.tstate, cl);
-        }
-    }
-
-    private native void setClassLoader(long tstate, ClassLoader cl);
-
-    /**
-     * Changes behavior of {@link #eval(String)}. Interactive mode can wait for
-     * further Python statements to be evaled, while non-interactive mode can
-     * only execute complete Python statements.
-     * 
-     * @deprecated This may be removed in a future version of Jep.
-     * 
-     * @param v
-     *            if the sub-interpreter should run in interactive mode
-     */
-    @Deprecated
-    public void setInteractive(boolean v) {
-        this.interactive = v;
-    }
-
-    /**
-     * Gets whether or not this sub-interpreter is interactive.
-     * 
-     * @deprecated This may be removed in a future version of Jep.
-     * 
-     * @return whether or not the sub-interpreter is interactive
-     */
-    @Deprecated
-    public boolean isInteractive() {
-        return this.interactive;
-    }
-
     @Override
     public void set(String name, Object v) throws JepException {
         isValidThread();
 
         if (v instanceof Class) {
             set(tstate, name, (Class<?>) v);
         } else if (v instanceof String) {
-            set(name, ((String) v));
+            set(tstate,name, ((String) v));
```

Comment:
I think we should just pass the Object to the native code and remove this if block. We used to have similar checks/conversions in many different places and the inconsistencies were a problem so I have been consolidating the checks ad this one should be the next to go.

The conversion code already has the same type of conversion in c so this is just redundant and it will simplify things to remove many of the native methods from this class.

Reply:
Are you saying to remove all these ifs and just have the code in the final else, i.e. `set(tstate, name, object)`?  No more instanceof and casting?  That would be most clean in my opinion.

Reply:
Yes, that is what is happening in the later patchsets and it is still passing tests on AppVeyor which is what I expected.